app/assets/javascripts/discourse/app/widgets/quick-access-bookmarks.js

```diff
@@ -42,13 +42,16 @@ createWidgetFrom(QuickAccessPanel, "quick-access-bookmarks", {
   },
 
   itemHtml(bookmark) {
+    // for topic level bookmarks we want to jump to the last unread post
+    // instead of the OP
+    let postNumber = bookmark.linked_post_number;
+    if (bookmark.for_topic) {
+      postNumber = bookmark.last_read_post_number + 1;
+    }
```

Comment:
```suggestion
    let postNumber;
    
    if (bookmark.for_topic) {
      postNumber = bookmark.last_read_post_number + 1;
    } else {
      postNumber = bookmark.linked_post_number;
    }
```

Reply:
How I wish for the ruby if assignment syntax in JS